lib/mapper.js

```diff
@@ -60,41 +60,40 @@ exports.page = function(page){
  * @api private
  */
 
-exports.track = function(track) {
+exports.track = function(track, settings) {
   // If the event name contains "unsubscribe", instead send an unsubscribe payload
   if (/[Uu]nsubscribe/.test(track.event())) {
     return reject({
-      auth_token: this.settings.authToken,
+      auth_token: settings.authToken,
       id: track.userId(),
       extras: {
         created_at: time(track.timestamp()),
         source: 'segment'
       }
     });
+  } 
+  var identity; 
+  if (typeof track.proxy('context.traits.email') !== 'undefined') {
```

Comment:
This can probably just be simplified to `if (track.email())`. If they explicitly pass email as `null`, we still probably don't want to send it.